src/java/htsjdk/variant/variantcontext/GenotypeLikelihoods.java

```diff
@@ -72,6 +72,13 @@
     */
     private final static GenotypeLikelihoodsAllelePair[] PLIndexToAlleleIndex = calculatePLcache(MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED);
 
+
+    /*
+    * a cache of the PL index to the anyploid alleles it represents over all possible numbers of alternate alleles
+    */
+    protected final static List<List<Integer>> plIndexToAlleleIndices = new ArrayList<List<Integer>>();
```

Comment:
scratch that. I got it wrong. it's not at all what I though...and I don't like it! The problem is that one cannot deal with two different ploidies at the same time since this is a singleton object I think that this should be encapsulated in a map (`Map<Integer,List<List<Integer>>>`) or an inner class so that one could have multiple instances of NonDiplodGenotypeLikelihoods with different ploidies...somewhat like LogMath in MathUtil.